src/field.rs

```diff
@@ -49,10 +68,16 @@ pub trait FieldElement:
     + Neg<Output = Self>
     + Display
     + From<<Self as FieldElement>::Integer>
+    + for<'a> TryFrom<&'a [u8], Error = FieldError>
+    // NOTE: Ideally this constraint would be `Into<[u8; Self::BYTES]>`, since that avoids a heap
```

Comment:
It is above the line being commented on. I would like for `FieldElement` to require `Into<[u8; Self::ENCODING.size]>` instead of `Into<Vec<u8>>`. I rewrote the comment (including removing the `:` to make this more clear.